pkg/postgres/pgadmin/admin_utils.go

```diff
@@ -107,20 +107,63 @@ func CheckIfDBExists(config *pgxpool.Config, dbName string) bool {
 	return exists
 }
 
+// AnalyzeDatabase - runs ANALYZE on the database named dbName
+func AnalyzeDatabase(config *pgxpool.Config, dbName string) error {
+	log.Debugf("Analyze - %q", dbName)
+
+	// Connect to different database for admin functions
+	connectPool := GetReplicaPool(config, dbName)
+	// Close the admin connection pool
+	defer connectPool.Close()
+
+	// Create a context with a timeout
+	ctx, cancel := context.WithTimeout(context.Background(), postgresQueryTimeout)
+	defer cancel()
+
+	_, err := connectPool.Exec(ctx, "ANALYZE")
```

Comment:
How long does this take if we restored a large database? If it takes long, postgresQueryTimeout may not be enough. And are we holding grpc call waiting for it?

Reply:
Right, analyzing could be a heavy operation if there is a lot of data.